src/signals/incident/components/form/MapSelectors/Caterpillar/CaterpillarLayer/CaterpillarLayer.tsx

```diff
@@ -18,33 +20,39 @@ import SelectContext from 'signals/incident/components/form/MapSelectors/Asset/c
 import featureSelectedMarkerUrl from 'shared/images/featureSelectedMarker.svg?url'
 
 import { featureToCoordinates } from 'shared/services/map-location'
-import ReportedLayer from '../../Asset/Selector/WfsLayer/ReportedLayer'
+import StatusLayer from '../../Asset/Selector/WfsLayer/StatusLayer'
 
 export const CaterpillarLayer: FC = () => {
   const { features } = useContext<FeatureCollection>(WfsDataContext)
   const { selection, meta, setItem, removeItem } = useContext(SelectContext)
 
   const getMarker = useCallback(
-    (feat: any) => {
+    (feat: any, reportedFeatureType, checkedFeatureType) => {
       const feature = feat as Feature
       const coordinates = featureToCoordinates(feature.geometry as Geometrie)
       // Caterpillar layer renders only a single feature type (oak tree)
       const featureType = meta.featureTypes.find(
         ({ typeValue }) => typeValue === 'Eikenboom'
       ) as FeatureType
-      const featureId = feature.properties[featureType.idField] as string
 
+      const featureId = feature.properties[featureType.idField] as string
       const isSelected = selection?.id === featureId
 
       const isReported = Boolean(
-        featureType.isReportedField &&
-          feature.properties[featureType.isReportedField] ===
-            featureType.isReportedValue
+        feature.properties[reportedFeatureType.isReportedField] ===
+          reportedFeatureType.isReportedValue
+      )
+      const isChecked = Boolean(
+        checkedFeatureType.isCheckedValues.includes(
+          feature.properties[checkedFeatureType.isCheckedField]
+        )
       )
```

Comment:
These particular checks happen in more places. Can we extract them and put them in one place?

Reply:
yes we can